server/src/main/java/io/crate/metadata/cluster/AlterTableClusterStateExecutor.java

```diff
@@ -236,31 +237,41 @@ private static String addExistingMeta(AlterTableRequest request, Map<String, Obj
         Map<String, Object> mappingDeltaAsMap = request.mappingDeltaAsMap();
         Map<String, Object> metaDelta = (Map<String, Object>) mappingDeltaAsMap.get("_meta");
         if (metaDelta != null) {
-            metaDelta.put("partitioned_by", currentMeta.getOrDefault("partitioned_by", List.of()));
-            metaDelta.putIfAbsent("primary_keys", currentMeta.get("primary_keys"));
+            if (currentMeta.get("partitioned_by") != null) {
+                metaDelta.put("partitioned_by", currentMeta.get("partitioned_by"));
```

Comment:
It would be possible to avoid the duplicate `get` calls, e.g.:

```suggestion
    var partitionedBy = currentMeta.get("partitioned_by");
    if (partitionedBy != null) {
        metaDelta.put("partitioned_by", partitionedBy);
    }
```

Reply:
ok Thanks, will make the changes